src/graphs/SMG.cc

```diff
@@ -38,6 +39,16 @@ const std::map<SMGValue, SMGEdgePointsToPtr>& SMG::GetPTEdges() const { return p
 
 const SMGEntitySet<const SMGEdgeHasValue>& SMG::GetHVEdges() const { return hv_edges_; }
 
+const SMGEntitySet<const SMGEdgeHasValue> SMG::GetHVEdges(
+    const SMGEdgeHasValueFilter& filter) const {
+  SMGEntitySet<const SMGEdgeHasValue> to_return;
+  std::copy_if(hv_edges_.begin(),
+               hv_edges_.end(),
+               std::inserter(to_return.set(), to_return.begin()),
+               filter);
+  return to_return;
+}
```

Comment:
Frankly, I do not remember anything related to this, which likely means it was not intentional. Reading the conversation, I think unifying them to a single place using copy_if is a right way to go.